tools/astarte_e2e/lib/astarte_e2e/client.ex

```diff
@@ -0,0 +1,575 @@
+#
+# This file is part of Astarte.
+#
+# Cospyright 2020 Ispirata Srl
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+defmodule AstarteE2E.Client do
+  alias Phoenix.Channels.GenSocketClient
+  alias Phoenix.Channels.GenSocketClient.Transport.WebSocketClient
+  alias AstarteE2E.{Utils, Config}
+
+  require Logger
+
+  @connection_backoff_ms 10_000
+  @connection_attempts 10
+
+  # API
+
+  @doc "Starts the client process."
+  @spec start_link(Config.client_options()) :: GenServer.on_start()
+  def start_link(opts) do
+    transport = WebSocketClient
```

Comment:
`@transport WebSocketClient` instead of using a var

Reply:
Or just pass `WebSocketClient` directly to the function